### DISCUSSION TITLE
Using Firebase Functions with Nuxt 3

### DISCUSSION THREAD

**Discussion by m6keller** [nuxt/nuxt#16403] [Questions]
Environment
Operating System: macOS 10.15.7 
Node Version:v16.14.2
Nuxt Version: 3.0.0-rc.2
Firebase: 9.7.0
firebase-admin: 10.2.0
firebase-functions: 3.21.0
firebase-functions-test: 0.3.3

In firebase.json the following config is set:
```JavaScript
{
"functions": { "source": ".output/server" }
}
```

I have a file under the "server" directory containing the following function:
```JavaScript
import * as functions from "firebase-functions";

export const helloWorld = functions.https.onRequest((request, response) => {
    functions.logger.info("Hello logs!", {structuredData: true});
    response.send("Hello from Firebase!");
});
```

When I run: 
```
NITRO_PRESET=firebase npm run build
firebase emulators:start --only functions
```
then go to my firebase emulator log, it does not show the new `helloWorld()` function being initialized. Also, when going to "http://localhost:5001/$PROJECTNAME/us-central1/helloWorld",  it returns "Function us-central1-helloWorld does not exist, valid functions are: us-central1-server" which suggests that my function has not been initialized.

Is there any way I can write firebase cloud functions in my Nuxt 3 app from files in my server directory?

I saw a similar discussion [here](https://github.com/nuxt/framework/discussions/2414) that said it was possible to change the nuxt.config.ts `functions` object between deploying functions,storage,firestore and deploying server and hosting. I am trying to write firebase functions solely in the "server" directory without creating a "functions" directory and the root of my project. Is this possible?

**Comment by rams3377** [comment]
Hello Everyone, I am wondering the same thing.
